tests/i18n/dates.py

```diff
@@ -6,21 +6,43 @@
 # or later license. See the LICENSE file for a copy of the license and the
 # AUTHORS file for copyright and authorship information.
 
+import pytest
 import time
 from datetime import datetime
 
+from babel import UnknownLocaleError, core as babel_core
 from babel.dates import format_timedelta
 
-from pootle.i18n.dates import localdate, timesince
+from django.utils.translation import get_language, override, to_locale
+
+from pootle.i18n.dates import timesince
 
 
 def test_local_date_timesince(settings):
+    for language in [get_language(), settings.LANGUAGE_CODE, 'en-us']:
+        try:
+            locale = babel_core.Locale.parse(to_locale(language))
+            break
+        except UnknownLocaleError:
+            continue
+
     timestamp = time.time() - 1000000
-    assert (
-        timesince(timestamp)
-        == format_timedelta(
-            datetime.now()
-            - datetime.fromtimestamp(timestamp),
-            locale=(
-                localdate.locale_code
-                or localdate.default_locale)))
+    timedelta = datetime.now() - datetime.fromtimestamp(timestamp)
+    assert timesince(timestamp) == format_timedelta(timedelta, locale=locale)
+
+
+@pytest.mark.parametrize('language,fallback', [
+    # Normal
+    ('af', 'af'),
+    ('en-za', 'en'),
+    ('en-us', 'en'),
+    # Missing in babel
+    ('son', 'en'),
```

Comment:
Bad one to test, this test will be bust as soon as `son` comes to babel. Rather us a really broken one that won't ever appear in CLDR